src/main/scala/org/broadinstitute/dsde/workbench/sam/service/PolicyEvaluatorService.scala

```diff
@@ -72,35 +72,36 @@ class PolicyEvaluatorService(
   def hasPermissionShallowCheck(resource: FullyQualifiedResourceId, action: ResourceAction, userId: WorkbenchUserId): IO[Boolean] = {
 
     for {
-      res <- resourceTypes.get(resource.resourceTypeName).traverse {
-        rt =>
-
-          val roleNamesWithAction = rt.roles.filter(_.actions.contains(action)).map(_.roleName)
-          val hasAction = directMemberHasActionOnResource(resource, roleNamesWithAction, action, userId)
+      result <- resourceTypes.get(resource.resourceTypeName).traverse {
+        resourceType =>
+          val roleNamesWithAction = resourceType.roles.filter(_.actions.contains(action)).map(_.roleName)
+          for {
+            hasAction <- directMemberHasActionOnResource(resource, roleNamesWithAction, action, userId)
+            authDomainCheckNotRequired = !resourceType.isAuthDomainConstrainable || !resourceType.actionPatterns.exists( ap => ap.authDomainConstrainable && ap.matches(action))
+            authDomainOK <- if (authDomainCheckNotRequired) IO.pure(true) else isSingleAuthDomainMember(resource, userId)
+          } yield {
+            hasAction && authDomainOK
+          }
+      }
+    } yield result.getOrElse(false)
+  }
 
-          // if the resource type OR action are NOT auth domain constrainable... just return
-          if (!rt.isAuthDomainConstrainable || !rt.actionPatterns.exists( ap => ap.authDomainConstrainable && ap.matches(action)) ) {
-            hasAction
+  private def isSingleAuthDomainMember(resource: FullyQualifiedResourceId, userId: WorkbenchUserId): IO[Boolean] = {
```

Comment:
thinking out loud... could there be a better name?  It's also true if this resource isn't actually constrained (regardless of if you're a member).  maybe isAuthDomainOkForSingleAuthDomainMember.... bleh.   Don't have a great suggestion for you